### PR TITLE
Create thin Docker Alpine skopeo image

### PR DESCRIPTION
**Pull request by alexei-led**
use Docker multi-stage build to create thin Docker image, based on Alpine linux and `skopeo` (compiled on alpine)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -68,6 +68,9 @@ binary-local:
 binary-local-static:
 	go build -ldflags "-extldflags \"-static\" -X main.gitCommit=${GIT_COMMIT}" -gcflags "$(GOGCFLAGS)" -tags "$(BUILDTAGS)" -o skopeo ./cmd/skopeo
 
+build-alpine-container:
```
Comment (rhatdan) [inline Makefile]: Would love to see this done with buildah....

File: Makefile [hunk 2]
```diff
@@ -68,6 +68,9 @@ binary-local:
...
+	docker build -t projectatomic/skopeo -f Dockerfile.ms .
+
```
Comment (rhatdan) [inline Makefile]: Curious: Why is it called .ms rather then .alpine?

File: Dockerfile.ms [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    libxml2 \
+    lz4-libs \
+    pcre \
+    sqlite-libs \
+    xz-libs \
+    zlib || apk update && apk upgrade
+
+RUN apk add --no-cache -X http://nl.alpinelinux.org/alpine/edge/testing ostree
+
+COPY --from=builder /go/src/github.com/projectatomic/skopeo/skopeo /usr/bin/skopeo
```
Comment (rhatdan) [inline Dockerfile.ms]: Add line.

File: Dockerfile.ms [hunk 4]
```diff
@@ -0,0 +1,62 @@
+#
+# ----- Go Builder Alpine Image ------
+#
+FROM golang:1.9-alpine AS builder
+
+RUN apk add --no-cache git bash || apk update && apk upgrade
```
Comment (mtrmac) [inline Dockerfile.ms]: What is the purpose of `|| app update && app upgrade` on every line?

That is, if I’m not mistaken, parsed as (`apk add` || `apk upgrade`) && (`apk upgrade`), i.e. running “update” if “add” fails, and running “upgrade” always (and failing only if “upgrade” fails).

File: Dockerfile.ms [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+RUN apk add --no-cache -X http://nl.alpinelinux.org/alpine/edge/testing ostree-dev
+
+# set working directory
+RUN mkdir -p /go/src/github.com/projectatomic/skopeo
+WORKDIR /go/src/github.com/projectatomic/skopeo
+
+# copy sources (including .git repo)
+COPY . .
+
+# run test and calculate coverage: skip for RELEASE
```
Comment (mtrmac) [inline Dockerfile.ms]: What does “skip for RELEASE” refer to?

File: Dockerfile.ms [hunk 6]
```diff
@@ -0,0 +1,62 @@
+#
+# ----- Go Builder Alpine Image ------
+#
+FROM golang:1.9-alpine AS builder
```
Comment (runcom) [inline Dockerfile.ms]: Is 1.9 GA?

**Comment by rhatdan** [comment]
@alexei-led Is this something you are still interested in, or should I close this?

**Comment by alexei-led** [comment]
Close if you like. Im OK with it

**Pull request closed without merging**